cmd/buildtools/openebs.go

```diff
@@ -78,3 +78,115 @@ var updateOpenEBSAddonCommand = &cli.Command{
 		return nil
 	},
 }
+
+type addonComponent struct {
+	name                        string
+	wolfiPackageName            string
+	getWolfiPackageName         func(upstreamVersion string) string
+	upstreamVersionMakefileVar  string
+	upstreamVersionFlagOverride string
+}
+
+var openebsComponents = []addonComponent{
+	{
+		name:                        "openebs-provisioner-localpv",
+		wolfiPackageName:            "dynamic-localpv-provisioner",
+		upstreamVersionMakefileVar:  "OPENEBS_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "openebs-version",
+	},
+	{
+		name:                        "openebs-linux-utils",
+		upstreamVersionMakefileVar:  "OPENEBS_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "openebs-version",
+	},
+	{
+		name:                        "openebs-kubectl",
+		getWolfiPackageName:         func(upstreamVersion string) string { return "kubectl-" + upstreamVersion + "-default" },
+		upstreamVersionMakefileVar:  "OPENEBS_KUBECTL_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "kubectl-version",
+	},
+}
+
+var updateOpenEBSImagesCommand = &cli.Command{
+	Name:      "openebs",
+	Usage:     "Updates the openebs images",
+	UsageText: environmentUsageText,
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "openebs-version",
+			Usage: "The version of openebs to use to determine image versions",
+		},
+		&cli.StringFlag{
+			Name:  "kubectl-version",
+			Usage: "The version of kubectl to use to determine image versions",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		logrus.Infof("updating openebs images")
+
+		if err := ApkoLogin(); err != nil {
+			return fmt.Errorf("failed to apko login: %w", err)
+		}
+
+		wolfiAPKIndex, err := GetWolfiAPKIndex()
+		if err != nil {
+			return fmt.Errorf("failed to get APK index: %w", err)
+		}
+
+		for _, component := range openebsComponents {
+			upstreamVersion, err := getAddonComponentUpstreamVersion(c, component)
+			if err != nil {
+				return fmt.Errorf("failed to get upstream version for %s: %w", component.name, err)
+			}
+
+			packageName := component.wolfiPackageName
+			if component.getWolfiPackageName != nil {
+				packageName = component.getWolfiPackageName(upstreamVersion)
+			}
+			packageVersion := upstreamVersion
+			if packageName != "" {
+				packageVersion, err = GetWolfiPackageVersion(wolfiAPKIndex, packageName, upstreamVersion)
+				if err != nil {
+					return fmt.Errorf("failed to get package version for %s: %w", component.name, err)
+				}
+			}
+
+			logrus.Infof("building and publishing %s=%s", component.name, packageVersion)
+
+			extraArgs := []string{}
+			if packageName != "" {
+				extraArgs = append(extraArgs, fmt.Sprintf("PACKAGE_NAME=%s", packageName))
```

Comment:
why not just always set this env var instead of passing it as an extra arg?

Reply:
i think you maybe misunderstood my point. I'm saying why not just set `PACKAGE_NAME` here similar to `PACKAGE_VERSION` instead of passing it as an extra arg?: https://github.com/replicatedhq/embedded-cluster/pull/794/files#diff-1d4bdab5fd148a75dead93758db106572c100974fc9a02efbb9125e9522e5db5R193

that way, the `PACKAGE_NAME` var would be automatically templated for other addons/packages as well when used in the apko file without having to explicitly pass it as an extra arg here.